src/test/groovy/org/shipkit/internal/gradle/plugin/PluginValidatorTest.groovy

```diff
@@ -0,0 +1,63 @@
+package org.shipkit.internal.gradle.plugin
+
+import testutil.PluginSpecification
+
+
+class PluginValidatorTest extends PluginSpecification {
+
+    private static final String META_INF_GRADLE_PLUGINS = 'src/main/resources/META-INF/gradle-plugins'
+    private static final String PLUGIN_PACKAGE = 'src/main/groovy/org/sipkit/gradle/'
+
+    def "validate plugin properties files"(propertiesFileName, className, extension) {
+        given:
+        project.file(META_INF_GRADLE_PLUGINS).mkdirs()
+        project.file(PLUGIN_PACKAGE).mkdirs()
+        def propertiesFile = project.file("$META_INF_GRADLE_PLUGINS/${propertiesFileName}.properties") << "implementation-class=org.shipkit.gradle.$className"
+        def pluginFile = project.file("$PLUGIN_PACKAGE/${className}.${extension}") << "TODO content"
+
+        when:
+        new PluginValidator().validate([pluginFile] as Set<File>, [propertiesFile] as Set<File>)
```

Comment:
Btw, have you checked Using `Set` instead of `def` to try to eliminate coercion?